spec/mongoid/relations/proxy_spec.rb

```diff
@@ -97,4 +97,28 @@ module Testable
       end
     end
   end
+
+  describe "equality" do
+    let(:messages) do
+      Person.create.messages
+    end
+
+    it "is #equal? to itself" do
+      expect(messages).to equal(messages)
+    end
+
+    it "is == to itself" do
+      expect(messages).to eq(messages)
+    end
+
+    it "is not #equal? to its target" do
+      expect(messages).to_not equal(messages.target)
+      expect(messages.target).to_not equal(messages)
+    end
+
+    it "is == to its target" do
+      expect(messages).to eq(messages.target)
+      expect(messages.target).to eq(messages)
+    end
```

Comment:
To increase clarity I would suggest to use the operator within the `expect()` call, for instance:

```
expect(messages == messages.target).to eq true
```